model/patch/cli_intent_test.go

```diff
@@ -111,6 +111,24 @@ func (s *CliIntentSuite) TestNewCliIntentRejectsInvalidIntents() {
 	s.Error(err)
 }
 
+func (s *CliIntentSuite) TestFindIntentSpecifically() {
+	intent, err := NewCliIntent(s.user, s.projectID, s.hash, s.module, "", s.description, true, s.variants, s.tasks, s.alias)
+	s.NoError(err)
+	s.NotNil(intent)
+	s.NoError(intent.Insert())
+
+	found, err := FindIntent(intent.ID(), intent.GetType())
+	s.NoError(err)
+	s.NotNil(found)
+
+	// the documents themselves aren't equal because of roundtrip db timestamp issues
```

Comment:
You can use [InDelta](https://godoc.org/github.com/stretchr/testify/assert#InDelta) to check the timestamps are close.

Reply:
Can you clarify what you meant by "roundtrip db timestamp issues?

Reply:
Basically, you'd think that you could just say "these things are equal" because we'd expect them to be, but the truth is that they're not because go timestamps are nanonseconds, and mongodb timestamps are milliseconds." 

It doesn't really matter, I just want to make sure that we have tests for the conversion (to and from) with some gaurd rails. I think the current implementation is (as written) sufficient, and I wanted to leave a note for future engineers to explain why it's not just "equals".

Reply:
Why not use .Round(time.Millisecond) then? I believe that also clears the monotonic clock as well.